Pod/Client/SBTUITunneledApplication.m

```diff
@@ -15,764 +15,347 @@
 {
+#if ENABLE_SWIZZLING
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         [XCTestCase loadSwizzles];
     });
+#endif
 }
 
-- (void)resetInternalState
+- (void)launchTunnelWithOptions:(nonnull NSArray<NSString *> *)options startupBlock:(nullable void (^)(void))startupBlock
 {
-    [self.bonjourBrowser stop];
-
-    self.launchArguments = self.initialLaunchArguments;
-    self.launchEnvironment = self.initialLaunchEnvironment;
-    self.startupBlock = nil;
-
-    self.bonjourName = [NSString stringWithFormat:@"com.subito.test.%d.%.0f", [NSProcessInfo processInfo].processIdentifier, (double)(CFAbsoluteTimeGetCurrent() * 100000)];
-    self.bonjourBrowser = [[NSNetService alloc] initWithDomain:@"local." type:@"_http._tcp." name:self.bonjourName];
-    self.bonjourBrowser.delegate = self;
-    self.connected = NO;
-    self.connectionPort = 0;
-    self.connectionTimeout = SBTUITunneledApplicationDefaultTimeout;
-}
+    NSMutableArray *launchArguments = [self.launchArguments mutableCopy];
+    [launchArguments addObjectsFromArray:options];
 
-- (void)terminate
-{
-    [self sendSynchronousRequestWithPath:SBTUITunneledApplicationCommandShutDown params:nil assertOnError:NO];
-    
-    [self resetInternalState];
-    
-    [super terminate];
-}
+    self.launchArguments = launchArguments;
 
-- (void)launchTunnel
-{
-    [self launchTunnelWithOptions:@[] startupBlock:nil];
+    [self launchTunnelWithStartupBlock: startupBlock];
 }
 
-- (void)launchTunnelWithStartupBlock:(void (^)(void))startupBlock;
-{
-    [self launchTunnelWithOptions:@[] startupBlock:startupBlock];
-}
+# pragma mark - SBTUITestTunnelClientDelegate
 
-- (void)launchTunnelWithOptions:(NSArray<NSString *> *)options startupBlock:(void (^)(void))startupBlock
+- (void)testTunnelClientIsReadyToLaunch:(nonnull SBTUITestTunnelClient *)sender
 {
-    NSMutableArray *launchArguments = [self.launchArguments mutableCopy];
-    [launchArguments addObjectsFromArray:options];
-    
-    [launchArguments addObject:SBTUITunneledApplicationLaunchSignal];
-    
-    if (startupBlock) {
-        [launchArguments addObject:SBTUITunneledApplicationLaunchOptionHasStartupCommands];
-    }
-    
-    self.startupBlock = startupBlock;
-    self.launchArguments = launchArguments;
-    
-    NSMutableDictionary<NSString *, NSString *> *launchEnvironment = [self.launchEnvironment mutableCopy];
-    launchEnvironment[SBTUITunneledApplicationLaunchEnvironmentBonjourNameKey] = self.bonjourName;
-    self.launchEnvironment = launchEnvironment;
-    
-    NSLog(@"[SBTUITestTunnel] Resolving bonjour service %@", self.bonjourName);
-    [self.bonjourBrowser resolveWithTimeout:self.connectionTimeout];
-    
     [self launch];
-    
-    [self waitForAppReady];
 }
 
-- (void)launchConnectionless:(NSString * (^)(NSString *, NSDictionary<NSString *, NSString *> *))command
+- (void)testTunnelClient:(nonnull SBTUITestTunnelClient *)sender didShutdownWithError:(NSError * _Nullable)error
```

Comment:
Given that nullability attributes are not required in implementation files I'd rather remove them altogether.

Reply:
Removed all nullability attributes from implementation files in most recent commit